core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java

```diff
@@ -0,0 +1,812 @@
+import io.questdb.mp.SCSequence;
+import io.questdb.mp.SPSequence;
+import io.questdb.mp.Sequence;
+import io.questdb.mp.WorkerPool;
+import io.questdb.std.CharSequenceObjHashMap;
+import io.questdb.std.Chars;
+import io.questdb.std.IntList;
+import io.questdb.std.LongList;
+import io.questdb.std.NumericException;
+import io.questdb.std.ObjList;
+import io.questdb.std.microtime.MicrosecondClock;
+import io.questdb.std.str.Path;
+import io.questdb.std.time.MillisecondClock;
+
+class LineTcpMeasurementScheduler implements Closeable {
+    private static final Log LOG = LogFactory.getLog(LineTcpMeasurementScheduler.class);
+    private static int REBALANCE_EVENT_ID = -1;
+    private static int INCOMPLETE_EVENT_ID = -2;
+    private final CairoEngine engine;
+    private final CairoSecurityContext securityContext;
+    private final CairoConfiguration cairoConfiguration;
+    private final MillisecondClock milliClock;
+    private RingQueue<LineTcpMeasurementEvent> queue;
+    private Sequence pubSeq;
+    private long nextEventCursor = -1;
+    private final CharSequenceObjHashMap<TableUpdateDetails> tableUpdateDetailsByTableName;
+    private final int[] loadByThread;
+
+    private final int nUpdatesPerLoadRebalance;
+    private final double maxLoadRatio;
+    private final int maxUncommittedRows;
+    private final long maintenanceJobHysteresisInMs;
+
+    private int nLoadCheckCycles = 0;
+    private int nRebalances = 0;
+
+    LineTcpMeasurementScheduler(CairoConfiguration cairoConfiguration, LineTcpReceiverConfiguration lineConfiguration, CairoEngine engine, WorkerPool writerWorkerPool) {
+        this.engine = engine;
+        this.securityContext = lineConfiguration.getCairoSecurityContext();
+        this.cairoConfiguration = cairoConfiguration;
+        this.milliClock = cairoConfiguration.getMillisecondClock();
+        tableUpdateDetailsByTableName = new CharSequenceObjHashMap<>();
+        loadByThread = new int[writerWorkerPool.getWorkerCount()];
+        int maxMeasurementSize = lineConfiguration.getMaxMeasurementSize();
+        int queueSize = lineConfiguration.getWriterQueueSize();
+        queue = new RingQueue<>(() -> {
+            return new LineTcpMeasurementEvent(maxMeasurementSize, lineConfiguration.getMicrosecondClock(), lineConfiguration.getTimestampAdapter());
+        }, queueSize);
+        pubSeq = new SPSequence(queueSize);
+
+        int nWriterThreads = writerWorkerPool.getWorkerCount();
+        Sequence[] subSeqById = new Sequence[nWriterThreads];
+        if (nWriterThreads > 1) {
+            FanOut fanOut = new FanOut();
+            for (int n = 0, sz = nWriterThreads; n < sz; n++) {
+                SCSequence subSeq = new SCSequence();
+                fanOut.and(subSeq);
+                subSeqById[n] = subSeq;
+                WriterJob writerJob = new WriterJob(n, subSeq);
+                writerWorkerPool.assign(n, writerJob);
+                writerWorkerPool.assign(n, writerJob::close);
```

Comment:
same minor thing here, creating identical lambdas in a loop

Reply:
These are not identical writerJob is scoped to the inside of the loop